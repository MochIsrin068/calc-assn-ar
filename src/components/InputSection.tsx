import { Checkbox, InputNumber } from "antd";

type TPropsInputSection = {
  inputValue: { value: number; checked: boolean };
  onChangeInputChecked: (newValue: boolean, index: number) => void;
  onChangeInputValue: (newValue: number, index: number) => void;
  index: number;
};

export default function InputSection({
  inputValue,
  onChangeInputChecked,
  onChangeInputValue,
  index,
}: TPropsInputSection) {
  return (
    <>
      <div className="flex justify-between bg-gray-400 p-3 mb-3 rounded-sm">
        <InputNumber
          value={inputValue.value}
          className="w-full mr-4 rounded-sm"
          type={"number"}
          placeholder="Masukkan angka"
          onChange={(value) => onChangeInputValue(value, index)}
          data-testid="number-input"
        />
        <Checkbox
          className="h-full"
          checked={inputValue.checked}
          onChange={(e) => onChangeInputChecked(e.target.checked, index)}
          data-testid="checkbox-input"
        />
      </div>
    </>
  );
}
