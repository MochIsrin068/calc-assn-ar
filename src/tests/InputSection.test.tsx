import React from "react";
import { render, screen } from "@testing-library/react";
import InputSection from "../components/InputSection";

const mockParams = {
  inputValue: { value: 10, checked: false },
  onChangeInputChecked: jest.fn(),
  onChangeInputValue: jest.fn(),
  index: 0,
};

describe("tests component InputSection", () => {
  it("Perform snapshots test InputSection", () => {
    const inputDefaultTree = render(<InputSection {...mockParams} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test input number placeholder and assertion value & type", () => {
    render(<InputSection {...mockParams} />);
    const item = screen.queryByTestId("number-input");
    expect(item).toBeInTheDocument();
    expect(item).toHaveAttribute("placeholder", "Masukkan angka");
    expect(item).toHaveAttribute("type", "number");
    expect(item).toHaveAttribute("value", "10");
  });

  it("test input checkbox is exist and assertion type", () => {
    render(<InputSection {...mockParams} />);
    const item = screen.queryByTestId("checkbox-input");
    expect(item).toBeInTheDocument();
    expect(item).toHaveAttribute("type", "checkbox");
  });
});
