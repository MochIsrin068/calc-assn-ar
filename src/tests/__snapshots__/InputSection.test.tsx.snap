// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests component InputSection Perform snapshots test InputSection 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between bg-gray-400 p-3 mb-3 rounded-sm"
      >
        <div
          class="ant-input-number w-full mr-4 rounded-sm"
        >
          <div
            class="ant-input-number-handler-wrap"
          >
            <span
              aria-disabled="false"
              aria-label="Increase Value"
              class="ant-input-number-handler ant-input-number-handler-up"
              role="button"
              unselectable="on"
            >
              <span
                aria-label="up"
                class="anticon anticon-up ant-input-number-handler-up-inner"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="up"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </span>
            </span>
            <span
              aria-disabled="false"
              aria-label="Decrease Value"
              class="ant-input-number-handler ant-input-number-handler-down"
              role="button"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-input-number-handler-down-inner"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-input-number-input-wrap"
          >
            <input
              aria-valuenow="10"
              autocomplete="off"
              class="ant-input-number-input"
              data-testid="number-input"
              placeholder="Masukkan angka"
              role="spinbutton"
              step="1"
              type="number"
              value="10"
            />
          </div>
        </div>
        <label
          class="ant-checkbox-wrapper h-full"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              data-testid="checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between bg-gray-400 p-3 mb-3 rounded-sm"
    >
      <div
        class="ant-input-number w-full mr-4 rounded-sm"
      >
        <div
          class="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled="false"
            aria-label="Increase Value"
            class="ant-input-number-handler ant-input-number-handler-up"
            role="button"
            unselectable="on"
          >
            <span
              aria-label="up"
              class="anticon anticon-up ant-input-number-handler-up-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="up"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </span>
          </span>
          <span
            aria-disabled="false"
            aria-label="Decrease Value"
            class="ant-input-number-handler ant-input-number-handler-down"
            role="button"
            unselectable="on"
          >
            <span
              aria-label="down"
              class="anticon anticon-down ant-input-number-handler-down-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class="ant-input-number-input-wrap"
        >
          <input
            aria-valuenow="10"
            autocomplete="off"
            class="ant-input-number-input"
            data-testid="number-input"
            placeholder="Masukkan angka"
            role="spinbutton"
            step="1"
            type="number"
            value="10"
          />
        </div>
      </div>
      <label
        class="ant-checkbox-wrapper h-full"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            data-testid="checkbox-input"
            type="checkbox"
            value=""
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
